Cards/CardMovementComponent.gd

```diff
@@ -28,14 +41,26 @@ func set_movement_state(new_state: Enums.CardMovementState):
 
 
 func _on_state_enter(state: Enums.CardMovementState):
-	pass
+	if _has_state(state):
```

Comment:
I'm not too up to date on what this does, quick explanation ? Since I see it's a repeating pattern between some functions

Reply:
You mean `_has_state`? This is just checking if the _state_mapping has a non-null value.

Reply:
no I meant what was coming after, sorry

Reply:
Yeah this is when the state initializes. This part gets called right when we enter a new state from set_movement_state. And init_state is just passing the state resource object stored here to the newly set state. Then when you change the state_properties in here, it will also update in the state itself